packages/bot/src/stores/flyout-store.js

```diff
@@ -0,0 +1,202 @@
+                const key = node.getAttribute('text') + index;
+
+                flyout_components.push(
+                    <div key={key} className='flyout__item-label'>
+                        { node.getAttribute('text') }
+                    </div>
+                );
+            } else if (tagName === 'BUTTON') {
+                const callbackKey = node.getAttribute('callbackKey');
+                const callback = Blockly.derivWorkspace.getButtonCallback(callbackKey) || (() => {});
+                const key = callbackKey + index;
+
+                flyout_components.push(
+                    <button
+                        key={key}
+                        className='flyout__button'
+                        onClick={(button) => {
+                            const flyout_button = button;
+
+                            // Workaround for not having a flyout workspace.
+                            flyout_button.targetWorkspace_ = Blockly.derivWorkspace;
+                            flyout_button.getTargetWorkspace = () => flyout_button.targetWorkspace_;
+
+                            callback(flyout_button);
+                        }}
+                    >
+                        { node.getAttribute('text') }
+                    </button>
+                );
+            }
+        });
+
+        this.flyout_content = flyout_components;
+        this.setVisibility(true);
+        this.setFlyoutWidth(xmlList);
+    }
+
+    /**
+     * Sets whether the flyout is visible or not.
+     * @param {boolean} is_visible
+     * @memberof FlyoutStore
+     */
+    @action.bound setVisibility(is_visible) {
+        this.is_visible = is_visible;
+
+        if (!is_visible) {
+            this.flyout_content = [];
+        }
+    }
+
+    /**
+     * Intialises a workspace unique to the passed block_node
+     * @param {Element} el_block_workspace Element where Blockly.Workspace will be mounted on
+     * @param {Element} block_node DOM of a Blockly.Block
+     * @memberof FlyoutStore
+     */
+    @action.bound initBlockWorkspace(el_block_workspace, block_node) {
+        const workspace = Blockly.inject(el_block_workspace, {
+            css   : false,
+            media : 'dist/media/',
+            move  : { scrollbars: false, drag: true, wheel: false },
+            sounds: false,
+        });
+
+        workspace.isFlyout = true;
+        workspace.targetWorkspace = Blockly.derivWorkspace;
+
+        const block = Blockly.Xml.domToBlock(block_node, workspace);
+
+        block.isInFlyout = true;
+
+        // Some blocks have hats, consider their height.
+        const extra_spacing = (block.startHat_ ? Blockly.BlockSvg.START_HAT_HEIGHT : 0);
+        const block_workspace_height = Number.parseInt(block_node.getAttribute('height')) + extra_spacing + 10;
+
+        // Update block workspace widths to accommodate block widths.
+        el_block_workspace.style.height = `${block_workspace_height}px`;
```

Comment:
@aaron-binary im not sure its the best way of handling it.we can use css-variables

Reply:
We don't know the dimensions of each block so we can't hardcode these values. Blocks can have different dimensions based on translations, or because they're user-generated ones like custom procedures and variables.